src/symmetric.rs

```diff
@@ -158,9 +159,10 @@ impl Keypair {
         system_parameters: &SystemParameters
     ) -> Keypair
     {
-        let a: Scalar = Scalar::hash_from_bytes::<Sha512>(&master_secret[..]);
-        let a0: Scalar = Scalar::hash_from_bytes::<Sha512>(a.as_bytes());
-        let a1: Scalar = Scalar::hash_from_bytes::<Sha512>(a0.as_bytes());
+        let h = Sha512::default().chain(&master_secret[..]);
+        let a: Scalar = Scalar::from_hash(h.clone().chain(b"a."));
+        let a0: Scalar = Scalar::from_hash(h.clone().chain(b"a0"));
+        let a1: Scalar = Scalar::from_hash(h.chain(b"a1"));
```

Comment:
Breaking change. The old hash functions reused keying material while the new functions are independent.